lib/buildFactory.js

```diff
@@ -136,29 +137,49 @@ class BuildFactory extends BaseFactory {
                     throw new Error('Job does not exist');
                 }
 
-                return getCommitInfo({ job, scm: this.scm, modelConfig })
-                    .then((data) => {
-                        // TODO: support matrix jobs
-                        const index = number.toString().split('.')[1] || 0;
-                        const permutation = job.permutations[index];
-
-                        modelConfig.sha = data.sha;
-
-                        if (data.decoratedCommit) {
-                            modelConfig.commit = data.decoratedCommit;
-                        }
-
-                        modelConfig.container = permutation.image;
-                        modelConfig.environment = permutation.environment;
-                        modelConfig.steps = permutation.commands.map(command => ({
-                            name: command.name,
-                            command: command.command
-                        }));
+                return Promise.all([
+                    job.pipeline,
+                    getCommitInfo({ job, scm: this.scm, modelConfig })
+                ]).then(([pipeline, data]) => {
+                    // TODO: support matrix jobs
+                    const index = number.toString().split('.')[1] || 0;
```

Comment:
No
We're not supporting matrix jobs yet, so we're just looking at the the first permutation object